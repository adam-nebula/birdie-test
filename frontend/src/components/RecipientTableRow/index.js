import moment from "moment";
import TableRow from "@material-ui/core/TableRow";
import TableCell  from "@material-ui/core/TableCell";

export default function RecipientTableRow({event, tableSchema}) {
    return (
        <TableRow key={event.id}>
            {tableSchema.map((field, index) => {
                return field.value.includes("timestamp") ? (
                    <TableCell align="right" key={event.id + "-" + index}>
                        {moment(event[field.value]).format('MMMM Do YYYY, h:mm:ss a')}
                    </TableCell>
                ) : (
                    <TableCell align="right" key={event.id + "-" + index}>
                        {event[field.value] != null ? event[field.value].toString() : ""}
                    </TableCell>
                )
            }
            )}
        </TableRow>
    )
}