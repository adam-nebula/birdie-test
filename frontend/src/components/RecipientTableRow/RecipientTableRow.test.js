import React from "react";
import renderer from 'react-test-renderer';
import RecipientTableRow from ".";

it("recipient table row renders", () => {

    const event = {id: "some_guid_that_shouldn't_display", 
    timestamp: "2019-05-01T07:26:06.684Z",
    event_type: "This_also_shouldn't_display",
    note: "some information",
    };

    const tableSchema = [{value: "note", label: "Note"}, {value: "timestamp", label: "Timestamp"}];

    const tree = renderer.create(
        <RecipientTableRow
        event={event}
        tableSchema={tableSchema}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})