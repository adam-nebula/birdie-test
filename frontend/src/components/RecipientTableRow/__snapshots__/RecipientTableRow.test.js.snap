// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipient table row renders 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    some information
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    May 1st 2019, 8:26:06 am
  </td>
</tr>
`;
